model/src/pyrenew/process/rtperiodicdiff.py

```diff
@@ -0,0 +1,270 @@
+# numpydoc ignore=GL08
+from typing import NamedTuple
+
+import jax.numpy as jnp
+from jax.typing import ArrayLike
+from pyrenew.metaclass import Model, RandomVariable, _assert_sample_and_rtype
+from pyrenew.process.firstdifferencear import FirstDifferenceARProcess
+
+
+class RtPeriodicDiffProcessProcessSample(NamedTuple):
+    """
+    A container for holding the output from `process.RtWeeklyDiffProcess.sample()`.
+
+    Attributes
+    ----------
+    rt : ArrayLike
+        The sampled Rt.
+    """
+
+    rt: ArrayLike | None = None
+
+    def __repr__(self):
+        return f"RtPeriodicDiffProcessProcessSample(rt={self.rt})"
+
+
+class RtPeriodicDiffProcess(Model):
+    r"""
+    Periodic Rt with autoregressive difference.
+
+    Notes
+    -----
+    This class samples a periodic Rt with autoregressive difference. The
+    mathematical formulation is given by:
+
+    .. math::
+        \log[\mathcal{R}^\mathrm{u}(t_3)] \sim \mathrm{Normal}\left(\log[\mathcal{R}^\mathrm{u}(t_2)] \
+            + \beta \left(\log[\mathcal{R}^\mathrm{u}(t_2)] - \
+             \log[\mathcal{R}^\mathrm{u}(t_1)]\right), \sigma_r \right)
+
+    where :math:`\mathcal{R}^\mathrm{u}(t)` is the periodic reproduction number
+    at time :math:`t`, :math:`\beta` is the autoregressive parameter, and
+    :math:`\sigma_r` is the standard deviation of the noise.
+    """
+
+    def __init__(
+        self,
+        n_obs: int,
+        data_starts: int,
+        period_size: int,
+        log_rt_prior: RandomVariable,
+        autoreg: RandomVariable,
+        sigma_r: RandomVariable,
+        site_name: str = "rt_periodic_diff",
+    ) -> None:
+        """
+        Default constructor for RtPeriodicDiffProcess class.
+
+        Parameters
+        ----------
+        n_obs : int
+            Number of observations.
```

Comment:
```suggestion
            Size of the returned sample.
```

Reply:
Solved in e0c64c0

Reply:
Suggest naming the argument `n_timepoints` for clarity